### PR TITLE
Patches group slice fix

### PR DESCRIPTION
**Pull request by benjaminpkane**
Fixes converting to patches in the App in group datasets when sidebar filters are present

* [x] e2e test
* [x] `fiftyone.server.view.get_view` unit test

### CODE REVIEW

File: e2e-pw/src/oss/poms/sidebar.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ export class SidebarPom {
   readonly sidebar: Locator;
   readonly asserter: SidebarAsserter;
 
-  constructor(page: Page) {
+  constructor(page: Page, eventUtils: any) {
```
Comment (sashankaryal) [inline e2e-pw/src/oss/poms/sidebar.ts:8]: can remove vestigial `eventUtils`

File: e2e-pw/src/oss/specs/groups/group-filter-toPatches.spec.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
+// create a grouped dataset
+// apply a sidebar filter
+// toPatches conversion
+// verify the result
```
Comment (sashankaryal) [inline e2e-pw/src/oss/specs/groups/group-filter-toPatches.spec.ts:4]: nit: may i suggest a multiline comment with bullets?

File: e2e-pw/src/oss/specs/groups/group-filter-toPatches.spec.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  gridActionsRow,
+  sidebar,
+  eventUtils,
+}) => {
+  await grid.assert.isEntryCountTextEqualTo("5 groups with slice");
+
+  // apply a sidebar filter
+  const entryExpandPromise = eventUtils.getEventReceivedPromiseForPredicate(
+    "animation-onRest",
+    () => true
```
Comment (sashankaryal) [inline e2e-pw/src/oss/specs/groups/group-filter-toPatches.spec.ts:77]: can safely remove this redundant second arg

File: e2e-pw/src/oss/specs/groups/group-filter-toPatches.spec.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  await gridActionsRow.clickToPatchesByLabelField("predictions");
+  await toPatchesRefresh;
+
+  // verify result:
+  await grid.assert.isEntryCountTextEqualTo("5 patches");
+
+  // not-carrot should not be in the sidebar filter anymore
+  const expandPromise = eventUtils.getEventReceivedPromiseForPredicate(
+    "animation-onRest",
+    () => true
```
Comment (sashankaryal) [inline e2e-pw/src/oss/specs/groups/group-filter-toPatches.spec.ts:101]: same as above

**Review by sashankaryal — APPROVED** [review]
looks good all in all, left some minor comments

**Review by lanzhenw — APPROVED** [review]

**Review by manivoxel51 — COMMENTED** [review]
🍨

**Pull request merged**